src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs

```diff
@@ -200,19 +200,48 @@ public void BuildPackage(PackCommandRunner runner)
             runner.BuildPackage();
         }
 
-        private string[] GetTargetPathsToAssemblies(IPackTaskRequest<IMSBuildItem> request)
+        private IEnumerable<OutputLibFile> InitLibFiles(IMSBuildItem[] libFiles)
         {
-            if (request.TargetPathsToAssemblies == null)
+            var assemblies = new List<OutputLibFile>();
+            if (libFiles == null)
             {
-                return new string[0];
+                return assemblies;
             }
 
-            return request.TargetPathsToAssemblies
-                .Where(path => path != null)
-                .Select(path => path.Trim())
-                .Where(path => path != string.Empty)
-                .Distinct()
-                .ToArray();
+
+            foreach (var assembly in libFiles)
+            {
+                var finalOutputPath = assembly.GetProperty("FinalOutputPath");
+                var targetPath = assembly.GetProperty("TargetPath");
+                var targetFramework = assembly.GetProperty("TargetFramework");
+
+                if (!File.Exists(finalOutputPath))
+                {
+                    throw new FileNotFoundException(string.Format(Strings.Error_FileNotFound, finalOutputPath));
```

Comment:
nit: set culture to current culture with string.Format, not needed with one param, but generally good to do for user facing errors.